index.html

```diff
@@ -0,0 +1,452 @@
+
+```graphql
+#::[Query with resolvers and realized selection sets]
+               # resolver              | realizes
+               # ---------------------------------------               
+query {        # (root)                -> self::Query
+    albums {   # albums::Query.albums  -> albums::Album
+      user {   # albums::Album.user    -> albums::User
+        name   # auth::User.name       -> String
+      }
+    }
+  }
+}
+```
+
+This query requires one portal: `name` is available on `self::User` but not on `albums::User`. Instead, it is provided only by `auth::User`. Since `name` is selected within an `albums::User`, the router must use a [portal](#portal-query) to acquire a selection on `auth::User` and resolve the field.
+
+### Free / Bound fields
+Free fields can be resolved by any subgraph which can realize a selection on the parent type. Bound fields can only be resolved by the subgraph they're bound to, which may require [porting](#portability) the selection set between graphs.
+
+### Portability
+A type is *portable to* a subgraph if it has one or more [`@cs_key`s](#cs_key) defined for that subgraph.
+
+```graphql
+#::[A query crossing subgraph boundaries]
+query {
+  me {
+    albums {
+      user {
+        name
+      }
+    }
+  }
+}
+```
+
+Incoming queries cross subgraph boundaries require "porting" wherever they select a field that is not resolvable by the subgraph of the realized type from which they are selected. In the above example, selecting `me.albums.user.name` crosses such a boundary, since `albums.user` is a selection on `albums::User` and `user.name` is only resolvable by `auth::User.name`.
+
+To resolve this query, `albums::User` must be *portable* to `auth::User`. That is, given a selection on `albums::User`, there must be some way to arrive at a selection on `auth::User`.
+
+If fields have multiple bound resolvers, routers **should** select the resolver which does not require crossing subgraph boundaries, if possible.
+
+If crossing subgraph boundaries is unavoidable, routers *must* use a [portal query](#portal-query) to re-enter the type in the destination subgraph.
+
+### Portal query
+In order to acquire a selection set on a entity within a subgraph, routers may issue a portal query to the subgraph:
+
+```graphql
+#::[Portal query]
+query {
+  _entities(representations: $representations) {
+    # ...selections...
+  }
+}
+```
+
+`representations` **must** be a list of input objects, each of which consist of the `__typename` of the object type or interface along with the result of fetching one of that subgraph's `key` fragments for the type.
```

Comment:
> along with the result of fetching one of that subgraph's `key` fragments for the type

Plus the fields specified in `@requires` on the parent field that returns the type.